examples/seq2seq/finetune_trainer.py

```diff
@@ -276,6 +276,7 @@ def main():
         # For convenience, we also re-save the tokenizer to the same directory,
         # so that you can share your model easily on huggingface.co/models =)
         if trainer.is_world_process_zero():
+            trainer.state.save_to_json(os.path.join(training_args.output_dir, "trainer_state.json"))
```

Comment:
Need to be saved for the tests. We can either save it here or have the tests return the trainer state instead of the output_dir. What do you prefer @sshleifer ?

Reply:
whatever you/ a user would prefer!

Reply:
Think having the state saved can't harm, so merging like this for now.